controllers/front/Payment.php

```diff
@@ -286,6 +287,17 @@ public function buildCCData($request, $payload)
             $request['paymentMethod']['holderName'] = $holderName;
         }
 
+        $shopperReference = $this->context->cart->id_customer;
+        if(!empty($shopperReference)) {
+            $request['shopperReference'] = $shopperReference;
+        }
+
+        //Oneclick data
+        if(!empty($payload['recurringDetailReference']) &&
+        $recurringDetailReference = $payload['recurringDetailReference']) {
```

Comment:
or maybe the whole recheck is not even important since the !empty part will check these so the `&&
        $recurringDetailReference = $payload['recurringDetailReference']` can just be removed